api/preprints/serializers.py

```diff
@@ -287,6 +330,18 @@ def update(self, preprint, validated_data):
             preprint.original_publication_date = validated_data['original_publication_date'] or None
             save_preprint = True
 
+        if 'has_prereg_links' in validated_data:
```

Comment:
What happens if the API request includes the already set values? Should these also check to see if the data has changed?

Reply:
I updated the update functions to exit immediately if value is unchanged.

Reply:
Sounds good. Maybe add one test that has all the fields filled out, but only one is different, and ensures that only the one proper log is added?